app/controllers/users_controller.rb

```diff
@@ -29,15 +30,34 @@ def update
   def create
     @user = User.new(user_params)
     if @user.save
+      session[:user_id] = @user.id
       flash[:notice] = "Welcome to the Meta Blog #{@user.username}"
       redirect_to articles_path
     else
       render 'new'
     end
   end
 
+  def destroy
+    @user.destroy
+    session[:user_id] = nil
```

Comment:
Question: what if `@user.destroy` has an error and the record is not destroyed? Your code here doesn't handle that case.

Your code here will show the notice and redirect but the record is not destroyed yet.

Fix it pls